### PR TITLE
[GPU] Fix a bug of strided_slice_ref kernel

### PR DESCRIPTION
**Pull request by jade-cho**
+ If begin, end and stride value as inputs of strided slice is negative, the index calculation for data read/write is going wrong.

### Tickets:
 - *127729*

### CODE REVIEW

**Comment by sammysun0711** [comment]
Verified that fixed Qwen with logits reduction opt on iGPU.

**Comment by isanghao** [comment]
no perf issue from dgpu daily test

**Review by isanghao — APPROVED** [review]
